tests/task.test.ts

```diff
@@ -517,11 +517,171 @@ describe.each([{ permission: 'Master' }, { permission: 'Admin' }])(
       const task = await client.waitForTask(enqueuedTask.taskUid)
-
       expect(task.type).toEqual(TaskTypes.TASK_CANCELATION)
       expect(task.details?.originalQuery).toContain('afterFinishedAt')
     })
 
+    // delete: uid
+    test(`${permission} key: Delete a task using the uid filter`, async () => {
+      const client = await getClient(permission)
+      const addDocuments = await client
+        .index(index.uid)
+        .addDocuments([{ id: 1 }])
+
+      const deleteTask = await client.deleteTasks({
+        uid: [addDocuments.taskUid],
+      })
+      const task = await client.waitForTask(deleteTask.taskUid)
+
+      expect(deleteTask.type).toEqual(TaskTypes.TASK_DELETION)
+      expect(task.details?.deletedTasks).toBeDefined()
+      await expect(client.getTask(addDocuments.taskUid)).rejects.toHaveProperty(
+        'code',
+        ErrorStatusCode.TASK_NOT_FOUND
+      )
+    })
+
+    // delete: indexUid
+    test(`${permission} key: Delete a task using the indexUid filter`, async () => {
+      const client = await getClient(permission)
+      const addDocuments = await client
+        .index(index.uid)
+        .addDocuments([{ id: 1 }])
+
+      const enqueuedTask = await client.deleteTasks({
+        indexUid: [index.uid],
+      })
+      const deleteTask = await client.waitForTask(enqueuedTask.taskUid)
+
+      expect(deleteTask.type).toEqual(TaskTypes.TASK_DELETION)
+      await expect(client.getTask(addDocuments.taskUid)).rejects.toHaveProperty(
+        'code',
+        ErrorStatusCode.TASK_NOT_FOUND
+      )
+    })
+
+    // delete: type
+    test(`${permission} key: Delete a task using the type filter`, async () => {
+      const client = await getClient(permission)
+
+      const enqueuedTask = await client.deleteTasks({
+        type: [
+          TaskTypes.DOCUMENTS_ADDITION_OR_UPDATE,
+          TaskTypes.DOCUMENT_DELETION,
+        ],
+      })
+      const deleteTask = await client.waitForTask(enqueuedTask.taskUid)
+
+      expect(deleteTask.type).toEqual(TaskTypes.TASK_DELETION)
+      expect(deleteTask.details?.originalQuery).toEqual(
+        'type=documentAdditionOrUpdate%2CdocumentDeletion'
+      )
+    })
+
+    // delete: status
+    test(`${permission} key: Delete a task using the status filter`, async () => {
+      const client = await getClient(permission)
+
+      const enqueuedTask = await client.deleteTasks({
+        status: [TaskStatus.TASK_ENQUEUED, TaskStatus.TASK_PROCESSING],
+      })
+      const task = await client.waitForTask(enqueuedTask.taskUid)
+
+      expect(task.type).toEqual(TaskTypes.TASK_DELETION)
+      expect(task.details?.originalQuery).toEqual(
+        'status=enqueued%2Cprocessing'
+      )
+    })
+
+    // delete: beforeEnqueuedAt
+    test(`${permission} key: Delete a task using beforeEnqueuedAt filter`, async () => {
+      const client = await getClient(permission)
+
+      const currentTimeStamp = Date.now()
+      const currentTime = new Date(currentTimeStamp)
```

Comment:
Why not just?
```suggestion
      const currentTime = new Date(Date.now())
```

Reply:
Or even just `Date.now()` directly?

Reply:
The date now returns a timestamp

> The static Date.now() method returns the number of milliseconds elapsed since January 1, 1970 00:00:00 UTC.

The Date constructor returns an instance of Date required by the function

> The Date() constructor can create a [Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) instance or return a string representing the current time.